bats/helpers/funding/onchain.bash

```diff
@@ -0,0 +1,84 @@
+CURRENT_FILE=${BASH_SOURCE:-bats/helpers/.}
+source "$(dirname "$CURRENT_FILE")/../_common.bash"
+
+fund_user_onchain() {
+  local token_name=$1
+  local wallet_id_name="$token_name.${2}_id" # btc_wallet or usd_wallet
+  local btc_amount_in_btc=${3:-"0.01"}
+
+  variables=$(
+    jq -n \
+    --arg wallet_id "$(read_value $wallet_id_name)" \
+    '{input: {walletId: $wallet_id}}'
+  )
+  exec_graphql "$token_name" 'on-chain-address-create' "$variables"
+  address="$(graphql_output '.data.onChainAddressCreate.address')"
+  [[ "${address}" != "null" ]] || exit 1
+
+  bitcoin_cli sendtoaddress "$address" "$btc_amount_in_btc"
+  bitcoin_cli -generate 2
+  retry 30 1 check_for_onchain_initiated_settled "$token_name" "$address"
+}
+
+check_for_onchain_initiated_settled() {
```

Comment:
I don't think the `check_xx` and `get_from_transaction_by_address` functions belong here under `funding`.

Reply:
FIxed in 9757bd4e56a6b4117e750b1782765b5ef35556e2